helm/provider.go

```diff
@@ -519,9 +519,6 @@ func (m *Meta) waitForTiller(o *installer.Options) error {
 }
 
 func (m *Meta) buildTunnel(d *schema.ResourceData) error {
-	if m.Settings.TillerHost != "" {
```

Comment:
This cannot be just removed as it also handles the case when the host is passed into provider configuration. I would suggest creating logic for building the tunnel only when it should be used, meaning when host is not provided from the configuration.